practice-app/server/src/api/controllers/user/enrollLesson.js

```diff
@@ -0,0 +1,43 @@
+import { User } from '../../../models/index.js';
+import { Lesson } from '../../../models/index.js';
+import { validateEnrollLesson } from '../../validators/user.validator.js';
+
+export default async (req, res) => {
+
+    const { error } = validateEnrollLesson(req.body);
+  
+    if (error) {
+      return res.status(400).json({ "resultMessage": error.details[0].message });
+    }
+  
+    const userId = req.body.user_id;
+    const lessonId = req.body.lesson_id;
+  
+    const lesson = await Lesson.findById(lessonId)
+    .catch(err => {
+      return res.status(500).json({ "resultMessage": err.message });
+    });
+  
+    if(!lesson) return res.status(400).json({"resultMessage": "The lesson with the given id could not be found"});
```

Comment:
Sorry I forgot to mention this one. I did the same mistake in my methods. This error status code should be 404 instead of 400. The next 400 codes should be 404 as well I guess.